src/main/java/io/fabric8/maven/docker/config/handler/compose/DockerComposeServiceWrapper.java

```diff
@@ -182,11 +182,22 @@ NetworkConfig getNetworkConfig() {
             NetworkConfig ret = new NetworkConfig(NetworkConfig.Mode.custom, custom);
             Object aliases = toJoin.get(custom);
             if (aliases != null) {
-                if (!(aliases instanceof List)) {
-                    throwIllegalArgumentException("'networks:' Aliases must be given as a list of string");
-                }
-                for (String alias : (List<String>) aliases) {
-                    ret.addAlias(alias);
+                if (aliases instanceof List) {
+                    for (String alias : (List<String>) aliases) {
+                        ret.addAlias(alias);
+                    }
+                } else if (aliases instanceof LinkedHashMap) {
```

Comment:
wouldn't it be better to check just for `Map` ?

Reply:
You are absolutly right